src/plugins/webcrack.ts

```diff
@@ -6,15 +6,51 @@ type File = {
   path: string;
 };
 
+/**
+ * Deobfuscate code using WebCrack and save the extracted result(s) into
+ * {@link outputDir}. Only files that are *created during this invocation* are
+ * returned.  Any previously–existing files in {@link outputDir} are ignored so
+ * that they are not processed a second time.
+ *
+ * The file that WebCrack calls `deobfuscated.js` is renamed to
+ * `[TIMESTAMP]_[ORIGINAL_FILENAME]_deobfuscated.js` so multiple runs can safely
+ * coexist in the same output directory without clashes.
+ */
 export async function webcrack(
   code: string,
-  outputDir: string
+  outputDir: string,
+  originalFilename: string
 ): Promise<File[]> {
+  const startTime = Date.now();
+
   const cracked = await wc(code);
   await cracked.save(outputDir);
 
-  const output = await fs.readdir(outputDir);
-  return output
-    .filter((file) => file.endsWith(".js"))
-    .map((file) => ({ path: path.join(outputDir, file) }));
+  const dirEntries = await fs.readdir(outputDir);
+  const timestamp = new Date().toISOString().replace(/[:.]/g, "-");
```

Comment:
[nit] I would probably put `startTime` and `timestamp` next to one another; and my initial thought when reading this was why not derive both from the same `Date` invocation.

Looking a bit closer, they're both formatted into different forms, so I understand why it was probably done this way; but if it's easy enough to, I wonder if it might be cleaner to align the times exactly.

One possible example:

```js
const runStart = new Date();
const timestamp = runStart.toISOString().replace(/[:.]/g, "-");

// ..snip..

if (stats.mtimeMs < runStart.getTime()) continue;
```